docs/config/plugins.md

```diff
@@ -0,0 +1,145 @@
+
+## Introduction
+
+<!-- See also https://github.com/wez/wezterm/commit/e4ae8a844d8feaa43e1de34c5cc8b4f07ce525dd -->
+
+A Wezterm plugin is a package of Lua files that provide
+some some predefined functionality not in the core product.
+
+A plugin is distributed via a Git URL.
+
+!!! Tip
+
+    Michael Brusegard maintains a [list of plugins](https://github.com/michaelbrusegard/awesome-wezterm)
+
+## Installing a Plugin
+
+Brief example:
+
+```lua
+local wezterm = require 'wezterm'
+local a_plugin = wezterm.plugin.require 'https://github.com/owner/repo'
+
+local config = wezterm.config_builder()
+
+a_plugin.apply_to_config(config)
+
+return config
+```
+
+The plugin URL must use the `HTTPS` or `file` [protocol](https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols).
+
+When Wezterm clones the repo into the runtime directory the default branch (probably `main`)
+is checked out and used as the plugin source.
+
+Plugins can be configured, for example:
+
+```lua
+local wezterm = require 'wezterm'
+local a_plugin = wezterm.plugin.require 'https://github.com/owner/repo'
+
+local config = wezterm.config_builder()
+
+local myPluginConfig = { enable = true, location = 'right' }
+
+a_plugin.apply_to_config(config, myPluginConfig)
+
+return config
+```
+
+!!! Note
+
+    Consult the README for a particular plugin to discover any specific configuration options.
+
+## Updating Plugins
+
+When changes are published to a plugin repository they are not updated in the local Wezterm instance.
+
+Run the command `wezterm.plugin.update_all()` to update all local plugins.
+
+!!! Tip
+
+    This can be run using the Lua REPL in [DebugOverlay](https://wezfurlong.org/wezterm/troubleshooting.html#debug-overlay).
+
+## Removing a Plugin
+
+When a plugin is first referenced, `wezterm.plugin.require()` will clone the repo if it doesn't already
+exist and store it in the runtime directory under `plugins/NAME` where
+`NAME` is derived from the repo URL.
+
+You can discover locations of the various plugins with `wezterm.plugin.list()`
+
+To remove the plugin simply delete the appropriate plugin directory.
+
+## Developing a Plugin
+
+1. Create a local development repo
+2. Add a file `plugin/init.lua`
+3. `init.lua` must return a module that exports an `apply_to_config`
+   function. This function must accept at least a config builder parameter, but may
+   pass other parameters, or a Lua table with a `config` field that maps
+   to a config build parameter
+4. Add any other Lua code needed to fulfil the plugin feature set.
+5. Add the plugin using a local file url e.g.
+   ```lua
+   local a_plugin = wezterm.plugin.require "file:///home/user/projects/myPlugin"
```

Comment:
There are some awesome tips thoughout the docs which area really great! 😄 
Just wanted to applaud you on adding them 😄